chmst/joomla-cms
# [4.0] Workflow Stage: Improve and fix delete and publish methods

Pull Request for Issue #21846 .

### Summary of Changes
Remove the method delete() from stageTable to stageModel.
Doing this some fixes have been added, comments and code have been imprved


### Testing Instructions
Make a workflow with several stages and transitions. 
Now try to change the status of one or more stages, thrash them, unpublish them.
Watch alerts, are they correkt?
Use the filter "trashed" and delete trashed stages.
Make sure that the transitions to and from the deletes stages are deleted too.



### Expected result



### Actual result



### Documentation Changes Required
